File: BaseSDK/Ubuntu/Samples/Base/DS86/DeviceIPAndSubnetMaskSet/DeviceIPAndSubnetMaskSet.cpp
#include <thread>
#include <iostream>
#include "Scepter_api.h"

using namespace std;

int main()
{
	cout << "---DeviceIPAndSubnetMaskSet---" << endl;

	uint32_t deviceCount;
	ScDeviceInfo* pDeviceListInfo = NULL;
	ScDeviceHandle deviceHandle = 0;
	ScStatus status = SC_OTHERS;

	status = scInitialize();
	if (status == ScStatus::SC_OK)
	{
		cout << "[scInitialize] success, ScStatus(" << status << ")." << endl;
	}
	else
	{
		cout << "[scInitialize] fail, ScStatus(" << status << ")." << endl;
		return -1;
	}

	status = scGetDeviceCount(&deviceCount, 3000);
	if (status == ScStatus::SC_OK)
	{
		cout << "[scGetDeviceCount] success, ScStatus(" << status << "). The device count is " << deviceCount << endl;
	}
	else
	{
		cout << "[scGetDeviceCount] fail, ScStatus(" << status << ")." << endl;
		return -1;
	}
	if (0 == deviceCount)
	{
		cout << "[scGetDeviceCount] scans for 3000ms and then returns the device count is 0. Make sure the device is on the network before running the samples." << endl;
		return -1;
	}

	pDeviceListInfo = new ScDeviceInfo[deviceCount];
	status = scGetDeviceInfoList(deviceCount, pDeviceListInfo);
	if (status == ScStatus::SC_OK)
	{
		cout << "[scGetDeviceInfoList] success, ScStatus(" << status << ").";
		if (SC_CONNECTABLE != pDeviceListInfo[0].status)
		{
			cout << " The first device [status]: " << pDeviceListInfo[0].status << " does not support connection." << endl;
			delete[] pDeviceListInfo;
			pDeviceListInfo = NULL;
			return -1;
		}
	}
	else
	{
		cout << "[scGetDeviceInfoList] fail, ScStatus(" << status << ")." << endl;
		delete[] pDeviceListInfo;
		pDeviceListInfo = NULL;
		return -1;
	}

	cout << " The first deviceInfo, <serialNumber>: " << pDeviceListInfo[0].serialNumber
		<< ", <ip>: " << pDeviceListInfo[0].ip << ", <status>: " << pDeviceListInfo[0].status << endl;

	status = scOpenDeviceBySN(pDeviceListInfo[0].serialNumber, &deviceHandle);
	if (status == ScStatus::SC_OK)
	{
		cout << "[scOpenDeviceBySN] success, ScStatus(" << status << ")." << endl;
		delete[] pDeviceListInfo;
		pDeviceListInfo = NULL;
	}
	else
	{
		cout << "[scOpenDeviceBySN] fail, ScStatus(" << status << ")." << endl;
		delete[] pDeviceListInfo;
		pDeviceListInfo = NULL;
		return -1;
	}

	status = scSetDeviceDHCPEnabled(deviceHandle, false);
	if (status == ScStatus::SC_OK)
	{
		cout << "[scSetDeviceDHCPEnabled] success, ScStatus(" << status << "). The device DHCP is close." << endl;
	}
	else
	{
		cout << "[scSetDeviceDHCPEnabled] fail, ScStatus(" << status << ")." << endl;
		return -1;
	}

	string strIP = "192.168.1.102";
	status = scSetDeviceIPAddr(deviceHandle, strIP.c_str(), strIP.length());
	if (status == ScStatus::SC_OK)
	{
		cout << "[scSetDeviceIPAddr] success, ScStatus(" << status << "). Set the device IP to " << strIP.c_str() << endl;
	}
	else
	{
		cout << "[scSetDeviceIPAddr] fail, ScStatus(" << status << ")." << endl;
		return -1;
	}

	string strSubMask = "255.255.255.0";
	status = scSetDeviceSubnetMask(deviceHandle, strSubMask.c_str(), strSubMask.length());
	if (status == ScStatus::SC_OK)
	{
		cout << "[scSetDeviceSubnetMask] success, ScStatus(" << status << "). Set the device subnet mask to " << strSubMask.c_str() << endl;
	}
	else
	{
		cout << "[scSetDeviceSubnetMask] fail, ScStatus(" << status << ")." << endl;
		return -1;
	}

	status = scRebootDevie(deviceHandle);
	if (status == ScStatus::SC_OK)
	{
		cout << "[scRebootDevie] success, ScStatus(" << status << ")." << endl;
	}
	else
	{
		cout << "[scRebootDevie] fail, ScStatus(" << status << ")." << endl;
		return -1;
	}

	cout << "Waiting for reboot." << endl;
	this_thread::sleep_for(chrono::seconds(10));
	cout << "Reboot done." << endl;

	status = scCloseDevice(&deviceHandle);
	if (status == ScStatus::SC_OK)
	{
		cout << "[scCloseDevice] success, ScStatus(" << status << ")." << endl;
	}
	else
	{
		cout << "[scCloseDevice] fail, ScStatus(" << status << ")." << endl;
		return -1;
	}

	status = scShutdown();
	if (status == ScStatus::SC_OK)
	{
		cout << "[scShutdown] success, ScStatus(" << status << ")." << endl;
	}
	else
	{
		cout << "[scShutdown] fail, ScStatus(" << status << ")." << endl;
		return -1;
	}
	cout << "---End---" << endl;

	return 0;
}

File: MultilanguageSDK/Python/Samples/DS77C/DeviceIPAndSubnetMaskSet/DeviceIPAndSubnetMaskSet.py
from pickle import FALSE, TRUE
import sys
currentPath =  sys.path[0]
pos = currentPath.find('Samples')
libpath = currentPath[:pos]
sys.path.append(libpath) #absolutely path

from API.ScepterDS_api import *
import time
print("---DeviceIPAndSubnetMaskSet---")
camera = ScepterTofCam()

camera_count = camera.scGetDeviceCount(3000)
print("[scGetDeviceCount] success, ScStatus(0), The device count is {}".format(camera_count))
if camera_count <= 0:
    print("[scGetDeviceCount] scans for 3000ms and then returns the device count is 0. Make sure the device is on the network before running the samples.")
    exit(1)

device_info=ScDeviceInfo()

if camera_count > 0:
    ret,device_infolist=camera.scGetDeviceInfoList(camera_count)
    if ret==0:
        device_info = device_infolist[0]
        print("[scGetDeviceInfoList] success, ScStatus({}). The first deviceInfo, <serialNumber>: {} , <ip>: {} , <status>: {}".format(ret, str(device_info.serialNumber.decode()), str(device_info.ip.decode()), str(device_info.status)))
        if  ScConnectStatus.SC_CONNECTABLE.value != device_info.status:
            print(" The first device [status]: {} does not support connection.".format(str(device_info.status)))
            exit(1)
    else:
        print("[scGetDeviceInfoList] fail, ScStatus({})".format(ret))
        exit(1) 
else: 
    print("there are no camera found")
    exit(1)

ret = camera.scOpenDeviceBySN(device_info.serialNumber)
if  ret == 0:
    print('[scOpenDeviceBySN] success ScStatus({}).'.format(str(ret)))
else:
    print('[scOpenDeviceBySN] fail ScStatus({}).'.format(str(ret)))
    exit(1)

'''Set the device as non-DHCP mode.'''
ret = camera.scSetDeviceDHCPEnabled(False)
if ret == 0:
    print('[scSetDeviceDHCPEnabled] success ScStatus({}). The device DHCP is close.'.format(str(ret)))
else:
	print('[scSetDeviceDHCPEnabled] fail ScStatus({}).'.format(str(ret)))
	exit(1)

'''Set the IP address of the device in non-DHCP mode.'''
strIP = r"192.168.1.102"
c_strIP = (c_char * 16)(*bytes(strIP, 'utf-8'))
ret = camera.scSetDeviceIPAddr(byref(c_strIP), 16)
if ret == 0:
    print('[scSetDeviceIPAddr] success ScStatus({}). Set the device IP to {}.'.format(str(ret), strIP))
else:
	print('[scSetDeviceIPAddr] fail ScStatus({}).'.format(str(ret)))
	exit(1)

'''Set the subnet mask of the device in non-DHCP mode.'''
strSubMask = r"255.255.255.0"
c_strSubMask= (c_char * 16)(*bytes(strSubMask, 'utf-8'))
ret = camera.scSetDeviceSubnetMask(byref(c_strSubMask), 16)
if ret == 0:
    print('[scSetDeviceSubnetMask] success ScStatus({}). Set the subnet mask to {}.'.format(str(ret), strSubMask))
else:
	print('[scSetDeviceSubnetMask] fail ScStatus({}).'.format(str(ret)))
	exit(1)

'''When the device is rebooted, the set IP and subnet mask take effect.'''
ret = camera.scRebootDevie()
if ret == 0:
    print('[scRebootDevie] success ScStatus({}).'.format(str(ret)))
else:
	print('[scRebootDevie] fail ScStatus({}).'.format(str(ret)))
	exit(1)

print("Waiting for reboot.")
time.sleep(10)
print("Rebort done.")

ret = camera.scCloseDevice()
if  ret == 0:
    print('[scCloseDevice] success ScStatus({}).'.format(str(ret)))
else:
    print('[scCloseDevice] fail ScStatus({}).'.format(str(ret)))
    exit(1)

exit(0)